From 10ddd66cd252b83f85f0e1680b38db30e0f24b9a Mon Sep 17 00:00:00 2001
From: Sidhanta Sahu <quic_sidhanta@quicinc.com>
Date: Thu, 9 Nov 2023 15:40:24 -0800
Subject: [PATCH] wifi: mac80211: Fix warn on during recovery with wds sta
 interface

Port change from patch:
	mac80211/patches/707-a-mac80211-framework-to-support-MLO-recovery.patch

Signed-off-by: Sidhanta Sahu <quic_sidhanta@quicinc.com>
---
 net/mac80211/util.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/net/mac80211/util.c b/net/mac80211/util.c
index 413b29a..739d398 100644
--- a/net/mac80211/util.c
+++ b/net/mac80211/util.c
@@ -2537,6 +2537,8 @@ static int ieee80211_reconfig_link_bss(struct ieee80211_sub_if_data *sdata,
 			  BSS_CHANGED_ARP_FILTER |
 			  BSS_CHANGED_IDLE;
 
+		ieee80211_vif_cfg_change_notify(sdata, changed);
+		changed = 0;
 		/* Re-send beacon info report to the driver */
 		if (link->u.mgd.have_beacon)
 			changed |= BSS_CHANGED_BEACON_INFO;
@@ -2545,7 +2547,7 @@ static int ieee80211_reconfig_link_bss(struct ieee80211_sub_if_data *sdata,
 			link_conf->protected_keep_alive)
 			changed |= BSS_CHANGED_KEEP_ALIVE;
 
-		ieee80211_vif_cfg_change_notify(sdata, changed);
+		ieee80211_link_info_change_notify(sdata, link, changed);
 		break;
 	case NL80211_IFTYPE_OCB:
 		changed |= BSS_CHANGED_OCB;
-- 
2.17.1

